callisto/delivery/models.py

```diff
@@ -157,7 +168,31 @@ def decrypted_report(self, key):
         Raises:
           CryptoError: If the key and saved salt fail to decrypt the record.
         """
-        return _decrypt_report(salt=self.salt, key=key, encrypted=self.encrypted)
+        iterations = None
+        harden_runtime = False
+        hasher = identify_hasher(self.encode_prefix)
+
+        if self.encode_prefix == '':
+            # this will only be used in the case of entries made before encode prefixes were used
+            iterations = ORIGINAL_KEY_ITERATIONS
+            salt = self.salt
+            if hasher.iterations > iterations:
+                harden_runtime = True
+        else:
+            salt = self.encode_prefix.rsplit('$', 1)[1]
+
+        # if pbkdf2 is used and the number of iterations changes
+        if self.encode_prefix != '' and hasher.algorithm == 'pbkdf2_sha256' and hasher.must_update(self.encode_prefix):
```

Comment:
don't think we need the algorithm check here. Can we subclass must_update to check for the empty string case rather than doing it here?